test/e2e/memory_reservation_locked_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	. "sigs.k8s.io/cluster-api/test/framework/ginkgoextensions"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+type MemoryReservationLockedSpecInput struct {
```

Comment:
```suggestion
type memoryReservationLockedSpecInput struct {
```

Reply:
I've been following the naming conventions from other tests, such as `hardware_upgrade_test.go`. Since there is no need to export anything from this test, would it be okay if I use lowercase for the function `VerifyMemoryReservationLockToMax()`? I would also like to rename the function to `verifyMemoryReservationLockedToMax` to better follow the convention.

Reply:
> I've been following the naming conventions from other tests, such as `hardware_upgrade_test.go`. Since there is no need to export anything from this test, would it be okay if I use lowercase for the function `VerifyMemoryReservationLockToMax()`? I would also like to rename the function to `verifyMemoryReservationLockedToMax` to better follow the convention.

yes